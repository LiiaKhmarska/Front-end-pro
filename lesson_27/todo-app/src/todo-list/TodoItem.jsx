export function TodoItem({todo, toggleTask, removeTask}) {
    return (
        <div className="item-container">
            <div className={todo.completed ? 'item completed' : 'item'}>
                <input type="checkbox" onClick={()=>toggleTask(todo.id)}/>
            {todo.task}
            </div>
            <div onClick={()=>removeTask(todo.id)}><i className="bi bi-trash"></i></div>
        </div>
        
    )
}