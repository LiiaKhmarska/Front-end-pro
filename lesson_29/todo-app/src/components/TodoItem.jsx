export function TodoItem({ todo }) {

    return (
        <div className="todo-container">
            {todo.content}
        </div>
    )
    
}