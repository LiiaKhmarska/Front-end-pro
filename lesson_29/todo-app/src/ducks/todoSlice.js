import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  todos: [{ id: 2147, content: "Make todo-list" }],
  todoCounter: 1,
};

const todoSlice = createSlice({
  name: "todo",
  initialState,
  reducers: {
    addTodo: (state, action) => {
      const newTodo = {
        id: Date.now(),
        content: action.payload.content,
      };
      state.todos.push(newTodo);
    },
    todoCount: (state) => {
      state.todoCounter = state.todoCounter + 1;
    },
  },
  selectors: {
    selectTodo: (state) => state.todos,
  },
});

export const { addTodo, todoCount } = todoSlice.actions;
export const { selectTodo } = todoSlice.selectors;

export default todoSlice.reducer;
