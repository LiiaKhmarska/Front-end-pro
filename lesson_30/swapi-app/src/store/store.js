import { configureStore } from "@reduxjs/toolkit";

import { dataUser } from "./userSlice";

export const store = configureStore({
  reducer: {
    user: dataUser,
  },
});
