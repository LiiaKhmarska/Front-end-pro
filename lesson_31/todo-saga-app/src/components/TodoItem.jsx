import { useDispatch } from 'react-redux';
import { DELETE_TODOS, FETCH_TODOS, PUT_TODOS } from '../ducks/todos/todos.actions';

export function TodoItem({ todo }) {
        const dispatch = useDispatch();

    function handlerRemoveTodo(id) {
        dispatch(DELETE_TODOS(id));
        // dispatch(FETCH_TODOS())
    }

    function handlerToggleStatus(todo) {

        todo = {...todo, completed: !todo.completed}
        dispatch(PUT_TODOS(todo));
    }

    return (
        <div className="todo-container">
            <input type='checkbox' checked={ todo.completed} onChange={()=>handlerToggleStatus(todo)}/>
            <div>{todo.content}</div>
            <button className="delete-btn" onClick={()=>handlerRemoveTodo(todo.id)}>‚ùå</button>
        </div>
    )
    
}