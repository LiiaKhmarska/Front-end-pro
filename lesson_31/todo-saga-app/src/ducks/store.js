import { configureStore } from "@reduxjs/toolkit";
import createSagaMiddleware from "redux-saga";
import userSagaReducer from "./todos/todos.slice.js";
import { rootSaga } from "./rootSaga";

const sagaMiddleware = createSagaMiddleware();

export const store = configureStore({
  reducer: {
    todosSaga: userSagaReducer,
  },
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware({ thunk: false, serialzableCheck: false }),
    sagaMiddleware,
  ],
});

sagaMiddleware.run(rootSaga);
