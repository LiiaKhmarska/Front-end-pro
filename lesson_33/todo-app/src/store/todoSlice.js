import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  todos: [{ id: 2147, content: "Make todo-list", checked: true }],
};

const todoSlice = createSlice({
  name: "todos",
  initialState,
  reducers: {
    addTodo: (state, action) => {
      const newTodo = {
        id: Date.now(),
        content: action.payload.content,
        checked: false,
      };
      state.todos.push(newTodo);
    },
    removeTodo: (state, action) => {
      const id = action.payload;
      state.todos = state.todos.filter((todo) => todo.id !== id);
    },
    toggleTodo: (state, action) => {
      const id = action.payload;
      const todo = state.todos.find((item) => item.id === id);
      if (todo) {
        todo.checked = !todo.checked;
      }
    },
  },
  selectors: {
    selectTodos: (state) => state.todos,
  },
});

export const { addTodo, todoCount, removeTodo, toggleTodo } = todoSlice.actions;
export const { selectTodos } = todoSlice.selectors;

export default todoSlice.reducer;
