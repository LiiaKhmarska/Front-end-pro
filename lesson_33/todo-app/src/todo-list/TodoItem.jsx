import { Checkbox,IconButton, ListItem, ListItemText, Tooltip, Typography } from "@mui/material"
import DeleteIcon from '@mui/icons-material/Delete';
import { useDispatch } from 'react-redux';
import {removeTodo,toggleTodo} from '../store/todoSlice'


export function TodoItem({ todo }) {
    const dispatch = useDispatch();

    function handleRemoveTodo(id) {
        dispatch(removeTodo(id));
    }

    function handleToggleTodo(id) {
        dispatch(toggleTodo(id))        
    }

    return (
        <>
            <ListItem sx={{ bgcolor: '#e3f2fd', borderRadius: 3, my: 1, px: 4, py: 2 }}>
                <Checkbox checked={todo.checked} onClick={()=>handleToggleTodo(todo.id)}/>
                <ListItemText >
                    <Typography sx={{
                        color: 'indigo', fontSize: 20,
                        textDecoration: todo.checked ? 'line-through' : 'none',
                    }}>
                        {todo.content}
                    </Typography>
                </ListItemText>
                <Tooltip title="Delete">
                    <IconButton edge='end' onClick={()=>{handleRemoveTodo(todo.id)}}>
                        <DeleteIcon />
                    </IconButton>
                </Tooltip>
            </ListItem>
            
        </>
    )
    
}