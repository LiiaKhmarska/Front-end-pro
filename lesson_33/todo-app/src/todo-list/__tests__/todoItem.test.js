import * as reduxHooks from "react-redux";
import { store } from "../../store/store";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import * as action from "../../store/todoSlice";
import { describe, expect, jest } from "@jest/globals";
import { TodoItem } from "../TodoItem";

describe("Test todo item", () => {
  it("should create todo item", () => {
    const todo = {
      id: 31081835,
      content: "Make todo-lisCheck todo item",
      checked: false,
    };
    const component = render(
      <Provider store={store}>
        <TodoItem todo={todo} />;
      </Provider>
    );
    expect(component).toMatchSnapshot();
    // expect(mockedDispatch).toHaveBeenCalled();
  });
});
