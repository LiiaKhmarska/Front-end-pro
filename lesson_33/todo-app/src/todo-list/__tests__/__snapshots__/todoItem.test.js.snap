// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test todo item should create todo item 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1d0g9p8-MuiListItem-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-vuhw6e-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <div
          class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-agyt4m-MuiTypography-root"
          >
            Make todo-lisCheck todo item
          </p>
        </div>
        <button
          aria-label="Delete"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
            />
          </svg>
        </button>
      </li>
      ;
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1d0g9p8-MuiListItem-root"
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-vuhw6e-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <div
        class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-agyt4m-MuiTypography-root"
        >
          Make todo-lisCheck todo item
        </p>
      </div>
      <button
        aria-label="Delete"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
          />
        </svg>
      </button>
    </li>
    ;
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
